src/lib.rs

```diff
@@ -547,15 +547,19 @@ fn render_multiplexor_signal(mut w: impl Write, signal: &Signal, msg: &Message)
             for multiplexer_index in multiplexer_indexes.iter() {
                 writeln!(
                     &mut w,
-                    "{idx} => {enum_name}::{multiplexed_wrapper_name}({multiplexed_name}{{ raw: self.raw }}),",
+                    "{idx} => Ok({enum_name}::{multiplexed_wrapper_name}({multiplexed_name}{{ raw: self.raw }})),",
                     idx = multiplexer_index,
                     enum_name = multiplex_enum_name(msg, signal)?,
                     multiplexed_wrapper_name = multiplexed_enum_variant_wrapper_name(*multiplexer_index),
                     multiplexed_name =
                         multiplexed_enum_variant_name(msg, signal, *multiplexer_index)?
                 )?;
             }
-            writeln!(&mut w, "_ => unreachable!(),")?;
+            writeln!(
+                &mut w,
+                "_ => Err(CanError::InvalidMultiplexor {{ message_id: {}}}),",
+                msg.message_id().0
```

Comment:
Hmm, looking at this again, do you think we should include the mulitplexor value? Can we always convert it to a u16 or something? Upside: Better debugging. Downside: Larger size of CanError type.

Reply:
Yes I think it's safe to say that it should be possible to convert into u16. I would probably even assume you'll never need more than u8. I think it certainly helps to add the value, otherwise it's hard to debug. So I think it's worth the downside. Went with u16 now, but like i said i can not imagine a use case that requires more than u8 and if there is such a use case that would at least to a compile time error with the generated code.